cpp/include/Ice/IconvStringConverter.h

```diff
@@ -181,30 +197,12 @@ IconvStringConverter<charT>::createDescriptors() const
 template<typename charT> std::pair<iconv_t, iconv_t>
 IconvStringConverter<charT>::getDescriptors() const
 {
-    void* val = pthread_getspecific(_key);
-    if(val != 0)
-    {
-        return *static_cast<std::pair<iconv_t, iconv_t>*>(val);
-    }
-    else
+    static thread_local DescriptorHolder descriptorHolder;
```

Comment:
I think we could do simpler and make the default ctor of DescriptorHolder call createDescritors();
```cpp
static thread_local DescriptorHolder descriptorHolder;
```

Then no need for the move-assignment etc.

Reply:
Fixed as suggested by I use a constructor with one parameter to pass the internal code.